syntax = "proto3";

package masterworker;

service MasterMachine{  
	rpc mapReduceRequest (MasterRequest) returns (WorkerResponse) {}
}

message FShard {
	int32 shard_id = 1;
	repeated ShardInfo info = 2;
}

message ShardInfo {
	string f_name = 1;
	int32 starts_at = 2;
	int32 ends_at = 3;	
}

message MapRequest {
	FShard shard = 1;
}

message ReduceRequest {
	int32 id = 1;
	FilePaths paths = 2;
	string out_dir = 3;
}

message FilePaths {
	repeated string f_name = 1;
}

message MasterRequest {
	MapRequest map_request = 1;
	ReduceRequest reduce_request = 2;
	string phase = 3;
	string user_id = 4;
}

message WorkerResponse {
	FilePaths paths = 1;
}